lib/sles4sap/sdaf_library.pm

```diff
@@ -445,6 +550,103 @@ sub get_tfvars_path {
     return $result;
 }
 
+=head2 prepare_tfvars_file
+
+    prepare_tfvars_file(deployment_type=>$deployment_type);
+
+B<$deployment_type>: Type of the deployment (workload_zone, sap_system, library... etc)
+
+Downloads tfvars template files from openQA data dir and places them into correct place within SDAF repo structure.
+Returns full path of the tfvars file.
+
+=cut
+
+sub prepare_tfvars_file {
+    my (%args) = @_;
+    my %tfvars_os_variable = (
+        deployer => 'deployer_parameter_file',
+        sap_system => 'sap_system_parameter_file',
+        workload_zone => 'workload_zone_parameter_file',
+        library => 'library_parameter_file'
+    );
+    my %tfvars_template_url = (
+        deployer => data_url('sles4sap/sdaf/DEPLOYER.tfvars'),
+        sap_system => data_url('sles4sap/sdaf/SAP_SYSTEM.tfvars'),
+        workload_zone => data_url('sles4sap/sdaf/WORKLOAD_ZONE.tfvars'),
+        library => data_url('sles4sap/sdaf/LIBRARY.tfvars')
+    );
+    croak 'Deplyoment type not specified' unless $args{deployment_type};
+    croak "Unknown deployment type: $args{deployment_type}" unless $tfvars_os_variable{$args{deployment_type}};
+
+    my $tfvars_file = get_os_variable($tfvars_os_variable{$args{deployment_type}});
+    my $retrieve_tfvars_cmd = join(' ', 'curl', '-v', '-fL', $tfvars_template_url{$args{deployment_type}}, '-o', $tfvars_file);
+
+    assert_script_run($retrieve_tfvars_cmd);
+    assert_script_run("test -f $tfvars_file");
+    replace_tfvars_variables($tfvars_file);
+    upload_logs($tfvars_file, log_name => "$args{deployment_type}.tfvars");
+    return $tfvars_file;
+}
+
+=head2 replace_tfvars_variables
+
+    replace_tfvars_variables();
+
+B<$deployment_type>: Type of the deployment (workload_zone, sap_system, library... etc)
+
+Replaces placeholder pattern B<%OPENQA_VARIABLE%> with corresponding OpenQA variable value.
+If OpenQA variable is not set, placeholder is replaced with empty value.
+
+=cut
+
+sub replace_tfvars_variables {
```

Comment:
This function has very limited functionality atm. I would leave refinement though out of scope of this PR  once requirements are more clear.